### ISSUE TITLE
dhcp_option_force in wrong dhcp interface section

### ISSUE THREAD

**Issue by flintt** [openNDS/openNDS#582]
I configured opennds to work in the `nds`  interface, but `dhcp_option_force` stay in the `lan` section. 
> uci export opennds

```
package opennds

config opennds
        option login_option_enabled '1'
        option gatewayinterface 'br-nds'
        option gatewayfqdn 'portal.local'
        list preauthenticated_users 'allow tcp port 22 to 192.168.2.1'
        list preauthenticated_users 'allow tcp port 80 to 192.168.2.1'
        option faskey '18bb53bb27adcefe0192718ada9e33ee665075f805ba87f281c5317abc18098e'
```
> uci export dhcp
```
package dhcp

config dnsmasq
        option local '/lan/'
        option domain 'lan'
        option readethers '1'
        option leasefile '/tmp/dhcp.leases'
        option localservice '1'
        option ednspacket_max '1232'
        option rebind_protection '0'
        option allservers '1'
        option noresolve '1'
        option strictorder '1'
        option boguspriv '0'
        option nonegcache '1'
        option cachesize '10000'
        list address '/portal.local/192.168.3.1'

config dhcp 'lan'
        option interface 'lan'
        option start '100'
        option limit '150'
        option dhcpv4 'server'
        option leasetime '1w'
        option force '1'
        option ra 'server'
        option dhcpv6 'hybrid'
        option dns_service '0'
        list dhcp_option_force '114,http://portal.local'

config dhcp 'wan'
        option interface 'wan'
        option ignore '1'

config odhcpd 'odhcpd'
        option maindhcp '0'
        option leasefile '/tmp/hosts/odhcpd'
        option leasetrigger '/usr/sbin/odhcpd-update'
        option loglevel '4'

config dhcp 'nds'
        option interface 'nds'
        option start '100'
        option limit '150'
        option leasetime '12h'
```

> uci export network
```
package network

config interface 'loopback'
        option device 'lo'
        option proto 'static'
        option ipaddr '127.0.0.1'
        option netmask '255.0.0.0'

config device
        option name 'br-lan'
        option type 'bridge'
        list ports 'eth0'

config interface 'lan'
        option device 'br-lan'
        option proto 'static'
        option netmask '255.255.255.0'
        option ip6assign '60'
        option delegate '0'
        option ipaddr '192.168.2.1'

config interface 'wan'
        option device 'eth1'
        option proto 'dhcp'
        option delegate '0'

config interface 'wan6'
        option device 'eth1'
        option proto 'dhcpv6'

config device
        option name 'eth1'
        option macaddr 'xxxx'

config device
        option type 'bridge'
        option name 'br-nds'

config interface 'nds'
        option proto 'static'
        option device 'br-nds'
        option ipaddr '192.168.3.1'
        option netmask '255.255.255.0'
```

> ndsctl status
```
==================
openNDS Status
====
Version: 10.1.3
Uptime: 48m 3s
Gateway Name: [ openNDS Node:186f2dcaa381  ]
Debug Level: [ 1 ]
Gateway FQDN: [ portal.local ]
Managed interface: br-nds
Upstream gateway(s) [ online:10.9.50.129,eth1  ]
MHD Server [ version 0.9.75 ] listening on: http://192.168.3.1:2050
Maximum Html Page size is [ 10240 ] Bytes
Preemptive Authentication is Enabled
Binauth Script: /usr/lib/opennds/binauth_log.sh
Preauth Script: /usr/lib/opennds/libopennds.sh
FAS: Secure Level 1, URL: http://portal.local:2050/opennds_preauth/
Client Check Interval: 15s
Rate Check Window: 2 check intervals (30s)
Preauthenticated Client Idle Timeout: 30m
Authenticated Client Idle Timeout: 120m
Download rate limit threshold (default per client): no limit
Upload rate limit threshold (default per client): no limit
Download quota (default per client): no limit
Upload quota (default per client): no limit
Total download: 267 kByte; average: 0.76 kbit/s
Total upload: 211 kByte; average: 0.60 kbit/s
====
Client authentications since start: 1
Current clients: 1

Client 0
  Client Type: cpd_can
  IP: 192.168.3.168 MAC: xxxxx
  Last Activity: Mon Apr 08 09:28:20 2024 (0s ago)
  Session Start: Mon Apr 08 08:41:21 2024 (46m 59s ago)
  Session End:   Tue Apr 09 08:41:21 2024 (23h 13m 1s left)
  Token: 08a1bef6
  State: Authenticated
  Download Rate Limit Threshold: not set
  Upload Rate Limit Threshold: not set
  Download quota: not set
  Upload quota: not set
  Download this session: 133 kB; Session average: 0.39 kb/s
  Upload this session: 148 kB; Session average: 0.43 kb/s

====
Trusted MAC addresses: none
Walled Garden FQDNs: none
Walled Garden Ports: none
========
```

**Comment by bluewavenet** [comment]
@flintt 

> Version: 10.1.3

The current version is 10.2.0
I suggest in the first instance that you upgrade.
Your issue was a known bug, and from memory a fix was implemented although I do not recall when.
I will check later.

**Comment by flintt** [comment]
On my other router running openwrt 23.05.0 with opennds 10.2.0 seems has the same problem

> ndsctl status
```
==================
openNDS Status
====
Version: 10.2.0
Uptime: 1m 32s
Gateway Name: [ openNDS Node:200cc845dbab  ]
Debug Level: [ 0 ]
Gateway FQDN: [ status.client ]
Managed interface: br-nds
All Upstream gateway(s) are offline or not connected [ offline ]
MHD Server [ version 0.9.75 ] listening on: http://192.168.4.1:2050
Maximum Html Page size is [ 10240 ] Bytes
Preemptive Authentication is Enabled
Binauth Script: /usr/lib/opennds/binauth_log.sh
ThemeSpec Core Library: /usr/lib/opennds/libopennds.sh
FAS: Secure Level 1, URL: http://status.client:2050/opennds_preauth/
Client Check Interval: 15s
Rate Check Window: 2 check intervals (30s)
Preauthenticated Client Idle Timeout: 30m
Authenticated Client Idle Timeout: 120m
Download rate limit threshold (default per client): no limit
Upload rate limit threshold (default per client): no limit
Download quota (default per client): no limit
Upload quota (default per client): no limit
Total download: 0 kByte; average: 0.00 kbit/s
Total upload: 55 kByte; average: 4.95 kbit/s
====
Client authentications since start: 0
Current clients: 0
====
Trusted MAC addresses:
none
====
Walledgarden FQDNs:
none

Walledgarden Ports:
all
====
Blocklist FQDNs:
none

Blocklist Ports:
all
========
```
> uci export dhcp
```
package dhcp

config dnsmasq
        option domainneeded '1'
        option localise_queries '1'
        option rebind_protection '0'
        option local '/lan/'
        option domain 'lan'
        option expandhosts '1'
        option cachesize '1000'
        option authoritative '1'
        option readethers '1'
        option leasefile '/tmp/dhcp.leases'
        option resolvfile '/tmp/resolv.conf.d/resolv.conf.auto'
        option localservice '1'
        option ednspacket_max '1232'
        list address '/status.client/192.168.4.1'

config dhcp 'lan'
        option interface 'lan'
        option start '100'
        option limit '150'
        option leasetime '12h'
        option dhcpv4 'server'
        option dhcpv6 'server'
        option ra 'server'
        option ra_slaac '1'
        list ra_flags 'managed-config'
        list ra_flags 'other-config'
        list dhcp_option_force '114,http://status.client'

config dhcp 'wan'
        option interface 'wan'
        option ignore '1'

config odhcpd 'odhcpd'
        option maindhcp '0'
        option leasefile '/tmp/hosts/odhcpd'
        option leasetrigger '/usr/sbin/odhcpd-update'
        option loglevel '4'

config dhcp 'nds'
        option interface 'nds'
        option start '100'
        option limit '150'
        option leasetime '12h'

config ipset 'nds_walledgarden'
        list name 'walledgarden'
        option table 'nds_filter'
        option table_family 'ip'

config ipset 'nds_blocklist'
        list name 'blocklist'
        option table 'nds_filter'
        option table_family 'ip'
```

**Comment by bluewavenet** [comment]
@flintt 
Looks like it is not fixed on 10.2.0 then. Something to check goes into the next release.
However this should not effect normal operation as it is only for rfc8910 clients. These are few and far between and will all fall back to CPD anyway.

**Comment by bluewavenet** [comment]
@flintt 
Fixed by commit https://github.com/openNDS/openNDS/pull/593/commits/569709a8678cdc9d5bd46fb9340704addaccc739
